compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -150,126 +151,114 @@ addTileAndDistributePasses(OpPassManager &funcPassManager,
-
-  // Verify that the translation info is using the right pipeline.
-  if (translationInfo.getDispatchLoweringPassPipeline() !=
-      IREE::Codegen::DispatchLoweringPassPipeline::CPUDoubleTilingExpert) {
-    return op->emitOpError("expected pipeline in translation_info to be ")
-           << stringifyEnum(IREE::Codegen::DispatchLoweringPassPipeline::
-                                CPUDoubleTilingExpert);
-  }
+LogicalResult verifyMultiTilingExpertPassPipelineConfig(
+    Operation *op, IREE::CPU::LoweringConfigAttr loweringConfig) {
 
-  if (tilingConfig.getNumTilingLevels() == 6) {
-    // TODO: update verification.
+  auto interfaceOp = dyn_cast_or_null<TilingInterface>(op);
+  if (!interfaceOp) {
     return success();
   }
 
-  if (tilingConfig.getNumTilingLevels() != 4) {
-    return op->emitOpError("expected four tiling levels, got ")
-           << tilingConfig.getNumTilingLevels();
+  // Collects parallel loops.
+  llvm::SmallDenseSet<unsigned> pLoopsSet;
+  for (auto [index, iteratorType] :
+       llvm::enumerate(interfaceOp.getLoopIteratorTypes())) {
+    if (iteratorType == utils::IteratorType::parallel) {
+      pLoopsSet.insert(index);
+    }
   }
 
-  auto interfaceOp = dyn_cast_or_null<TilingInterface>(op);
-  if (interfaceOp) {
-    llvm::SmallDenseSet<unsigned> pLoopsSet;
-    for (auto [index, iteratorType] :
-         llvm::enumerate(interfaceOp.getLoopIteratorTypes())) {
-      if (iteratorType == utils::IteratorType::parallel) {
-        pLoopsSet.insert(index);
-      }
+  for (int i = 0, e = IREE::CPU::TilingLevel::MaxNumTileLevels; i < e; ++i) {
+    if (!loweringConfig.hasTilingLevel(i)) {
+      continue;
     }
 
-    SmallVector<int64_t> secondLevelTileSizes;
-    std::tie(secondLevelTileSizes, std::ignore) =
-        tilingConfig.getVectorCommonParallelSizes();
-    for (auto [index, tileSize] : llvm::enumerate(secondLevelTileSizes)) {
-      if (tileSize != 0 && !pLoopsSet.contains(index)) {
-        return op->emitOpError(
-                   "expected only parallel dims to be set in the second tiling "
-                   "level, got ")
-               << index << "-th tile size set";
+    auto level = static_cast<IREE::CPU::TilingLevel>(i);
+    auto tilingLevelAttr = cast<IREE::Codegen::LoweringConfigTilingLevelAttr>(
+        loweringConfig.getTilingLevelAttr(level));
+    switch (level) {
+    case IREE::CPU::TilingLevel::DistributionTiles:
+    case IREE::CPU::TilingLevel::CacheParallelTiles:
+    case IREE::CPU::TilingLevel::VectorCommonParallelTiles:
+    case IREE::CPU::TilingLevel::VectorInnerParallelTiles: {
+      for (auto [index, tileSize] :
+           llvm::enumerate(tilingLevelAttr.getSizes())) {
+        if (tileSize != 0 && !pLoopsSet.contains(index)) {
+          return op->emitOpError(
+                     "expected only parallel dims to be set in the ")
+                 << IREE::CPU::getTilingLevelName(level)
+                 << " tiling level, got " << index << "-th tile size set";
```

Comment:
[nit] `"-th"` is a bit off when you get e.g. "first" or "second". You can work around it like this:
```suggestion
                 << " tiling level, but tile size at index (" << index << ") was also set";
```